model_compression_toolkit/data_generation/pytorch/model_info_exctractors.py

```diff
@@ -12,62 +12,61 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # ==============================================================================
-from typing import Type, Dict, Tuple
+from typing import Type, Dict, Tuple, List
 import torch
 from torch import Tensor
 from torch.nn import Module
 
-from model_compression_toolkit.data_generation.common.model_info_exctractors import OrigBNStatsHolder, \
+from model_compression_toolkit.core.pytorch.utils import get_working_device
+from model_compression_toolkit.data_generation.common.model_info_exctractors import OriginalBNStatsHolder, \
     ActivationExtractor
-from model_compression_toolkit.data_generation.pytorch.constants import DEVICE, IMAGE_INPUT
+from model_compression_toolkit.data_generation.pytorch.constants import IMAGE_INPUT, NUM_INPUT_CHANNELS
 
 
-class PytorchOrigBNStatsHolder(OrigBNStatsHolder):
+class PytorchOriginalBNStatsHolder(OriginalBNStatsHolder):
     """
     Holds the original batch normalization (BN) statistics for a model.
     """
     def __init__(self,
                  model: Module,
-                 bn_layer_types: Type[list],
-                 eps=1e-6):
+                 bn_layer_types: List):
         """
-        Initializes the PytorchOrigBNStatsHolder.
+        Initializes the PytorchOriginalBNStatsHolder.
 
         Args:
             model (Module): The PyTorch model.
-            bn_layer_types (Type[list]): List of batch normalization layer types.
-            eps (float): Epsilon value for numerical stability.
+            bn_layer_types (List): List of batch normalization layer types.
         """
-        super(PytorchOrigBNStatsHolder, self).__init__(model, bn_layer_types, eps)
+        super(PytorchOriginalBNStatsHolder, self).__init__(model, bn_layer_types)
 
     def get_bn_params(self,
                       model: Module,
-                      bn_layer_types: Type[list]) -> Dict[str, Tuple[Tensor, Tensor, Tensor]]:
+                      bn_layer_types: List) -> Dict[str, Tuple[Tensor, Tensor, Tensor]]:
         """
         Get the batch normalization parameters (mean and variance) for each batch normalization layer in the model.
 
         Args:
             model (Module): The PyTorch model.
-            bn_layer_types (Type[list]): List of batch normalization layer types.
+            bn_layer_types (List): List of batch normalization layer types.
 
         Returns:
             dict: Dictionary mapping batch normalization layer names to their parameters.
         """
         bn_params = {}
         # Assume the images in the dataset are normalized to be 0-mean, 1-variance
-        imgs_mean = torch.zeros(1, 3).to(DEVICE)
-        imgs_var = torch.ones(1, 3).to(DEVICE)
+        imgs_mean = torch.zeros(1, NUM_INPUT_CHANNELS).to(get_working_device())
```

Comment:
it's not very important but you can assign "get_working_device()" to a variable and use the variable instead calling to the function multiple times

Reply:
changed it in all the code